keyboards/adpenrose/kintsugi/keymaps/default/keymap.c

```diff
@@ -0,0 +1,105 @@
+          KC_LSFT,          KC_Z,    KC_X,    KC_C,   KC_V,   KC_B,   KC_N,   KC_M,   KC_COMM, KC_DOT,  KC_SLSH,  KC_RSFT,   KC_UP,    KC_DEL,
+          KC_LCTL, KC_LGUI, KC_LALT,                        KC_SPC,                   KC_RALT, MO(1),           KC_LEFT, KC_DOWN,  KC_RIGHT
+    ),
+    [1] = LAYOUT_65_ansi_blocker(
+      _______,    KC_F1,      KC_F2,      KC_F3,      KC_F4,     KC_F5,     KC_F6,     KC_F7,     KC_F8,     KC_F9,     KC_F10,     KC_F11,   KC_F12,  KC_DEL,
+      _______,  RGB_TOG,    RGB_VAI,    RGB_VAD,    RGB_MODE_FORWARD,   _______,   _______,   _______,   _______,   _______,    _______,    _______,  _______, _______,
+      RESET,    _______,    _______,    _______,    _______,   _______,   _______,   _______,   _______,   _______,    _______,    _______,  _______,            KC_MPLY,
+      _______,              _______,    _______,    _______,   _______,   _______,   _______,   _______,   _______,    _______,    _______,  _______, _______,   _______,
+      _______,  _______,    _______,                                        KC_SPC,                        _______,    _______,              _______,  _______,  _______
+    ),
+};
+
+/* Encoder */
+#ifdef ENCODER_ENABLE
+bool encoder_update_user(uint8_t index, bool clockwise) {
+/* The switch case allows for different encoder mappings on different layers, "default" map gets applied for all unspecified layers */
+    switch(get_highest_layer(layer_state)){
+        case 1:
+            if (clockwise) {
+                tap_code(KC_MNXT);
+            } else {
+                tap_code(KC_MPRV);
+            }
+            break;
+        default:
+            if (clockwise){
+                tap_code(KC_VOLU);
+            } else{
+                tap_code(KC_VOLD);
+            }
+            break;
+        }
+    return true;
+}
+#endif
+
+/* Rotation of the OLED: */
+#ifdef OLED_ENABLE
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+    return OLED_ROTATION_270;
+}
+
+/* Kintsugi logo render: */
+static void render_logo(void) {
+    static const char PROGMEM logo_1[] = {
+        0x83, 0x84, 0x85, 0x86, 0x87, 0xA3, 0xA4, 0xA5, 0xA6, 0xA7, 0xC3, 0xC4, 0xC5, 0xC6, 0xC7, 0x00
+    };
+    static const char PROGMEM logo_2[] = {
+        0x88, 0x89, 0x8A, 0x8B, 0x8C, 0xA8, 0xA9, 0xAA, 0xAB, 0xAC, 0xC8, 0xC9, 0xCA, 0xCB, 0xCC, 0x00
+    };
+    static const char PROGMEM logo_3[] = {
+        0x8D, 0x8E, 0x8F, 0x90, 0x91, 0xAD, 0xAE, 0xAF, 0xB0, 0xB1, 0xCD, 0xCE, 0xCF, 0xD0, 0xD1, 0x00
+    };
+    oled_set_cursor(1,3);
+    oled_write_P(logo_1, false);
+    oled_set_cursor(1,7);
+    oled_write_P(logo_2, false);
+    oled_set_cursor(1,11);
+    oled_write_P(logo_3, false);
+}
+
+/* Function that renders stuff on the oled: */
+bool oled_task_user(void) {
+    /* Function that renders the kintsugi logo in the desired order. */
+    render_logo();
+    return false;
+}
+#endif
```

Comment:
```suggestion
```

Reply:
Removing that portion of code would make the OLED and encoder features useless. Did you mean to mark that as a suggested change?

Reply:
Yes.  The code should go into the keyboard's c file, which I do have below. 

This would allow for the keyboard to have a default behavior without the keymap defining it.  This is ESPECIALLY useful for anyone wanting to compile from the QMK Configurator.